src/components/TeacherComponents/LessonOverviewHeaderDownloadAllButton/LessonOverviewHeaderDownloadAllButton.tsx

```diff
@@ -37,6 +38,10 @@ export const LessonOverviewHeaderDownloadAllButton: FC<
     ? "downloads-auth"
     : "downloads";
 
+  const { user } = useFeatureFlaggedClerk().useUser();
+
+  const displaySignInMessage = !user || !user?.publicMetadata?.owa?.isOnboarded;
```

Comment:
This isn't quite right. It will always be displayed for users without the feature flag as their `user` will be `undefined`

When Clerk loads and there is no logged in user the value will transition to `null` which we can use to detect if Clerk is enabled.

```suggestion
  const { user } = useFeatureFlaggedClerk().useUser();

  const displaySignInMessage =
    user === null || (user && !user.publicMetadata?.owa?.isOnboarded);
```

There is also `isLoaded` on `useUser` which will always be `false` when the feature flag is off so we could look for that too.

```suggestion
  const { user, isLoaded } = useFeatureFlaggedClerk().useUser();

  const displaySignInMessage =
    isLoaded && !user?.publicMetadata?.owa?.isOnboarded;
```

Reply:
Amended to utilise the isLoaded value